src/encode_lib_log_parser.py

```diff
@@ -0,0 +1,624 @@
+                result[key] = float(result[key])
+    return result
+
+MAP_KEY_DESC_FRIP_QC = {
+    'frip' : 'Fraction of Reads in Peaks',
+}
+
+def parse_frip_qc(txt):
+    result = OrderedDict()
+    if not txt: return result
+    with open(txt, 'r') as f:
+        frip = f.readlines()[0].strip()
+    result['frip'] = float(frip)
+    return result
+
+MAP_KEY_DESC_ANNOT_ENRICH_QC = {
+    'fri_dhs' : 'Fraction of Reads in universal DHS regions',
+    'fri_blacklist' : 'Fraction of Reads in blacklist regions',
+    'fri_prom' : 'Fraction of Reads in promoter regions',
+    'fri_enh' : 'Fraction of Reads in enhancer regions',
+}
+
+def parse_annot_enrich_qc(txt):
+    result = OrderedDict()
+    if not txt: return result
+    with open(txt, 'r') as fp:
+        lines = fp.read().strip('\n').split('\n')
+    for line in lines:
+        key, reads, frac = line.split('\t')
+        frac = to_float(frac)
+        if key == 'fraction_of_reads_in_universal_DHS_regions':
+            result['fri_dhs'] = frac
+        elif key == 'fraction_of_reads_in_blacklist_regions':
+            result['fri_blacklist'] = frac
+        elif key == 'fraction_of_reads_in_promoter_regions':
+            result['fri_prom'] = frac
+        elif key == 'fraction_of_reads_in_enhancer_regions':
+            result['fri_enh'] = frac
+        else:
+            raise ValueError(
+                'Wrong line in annot_enrich QC file')
+    return result
+
+MAP_KEY_DESC_PICARD_EST_LIB_SIZE_QC = {
+    'picard_est_lib_size' : 'Estimated library size by Picard tools',
+}
+
+def parse_picard_est_lib_size_qc(txt):
+    result = OrderedDict()
+    if not txt: return result
+    with open(txt, 'r') as f:
+        val = f.readlines()[0].strip()
+    result['picard_est_lib_size'] = float(val)
+    return result
+
+MAP_KEY_DESC_TSS_ENRICH_QC = {
+    'tss_enrich' : 'TSS enrichment',
+}
+
+def parse_tss_enrich_qc(txt):
+    result = OrderedDict()
+    if not txt: return result
+    with open(txt, 'r') as f:
+        val = f.readlines()[0].strip()
+    result['tss_enrich'] = float(val)
+    return result
+
+MAP_KEY_DESC_NUCLEOSOMAL_QC = {
+    'frac_reads_in_nfr' : 'Fraction of reads in NFR',
+    'frac_reads_in_nfr_qc_pass' : 'Fraction of reads in NFR (QC pass)',
+    'frac_reads_in_nfr_qc_reason' : 'Fraction of reads in NFR (QC reason)',
+    'nfr_over_mono_nuc_reads' : 'NFR / mono-nuc reads',
+    'nfr_over_mono_nuc_reads_qc_pass' : 'NFR / mono-nuc reads (QC pass)',
+    'nfr_over_mono_nuc_reads_qc_reason' : 'NFR / mono-nuc reads (QC reason)',
+    'nfr_peak_exists' : 'Presence of NFR peak',
+    'mono_nuc_peak_exists' : 'Presence of Mono-Nuc peak',
+    'di_nuc_peak_exists' : 'Presence of Di-Nuc peak',
+}
+
+def parse_nucleosomal_qc(txt):
+    result = OrderedDict()
+    if not txt: return result
+    with open(txt, 'r') as fp:
+        lines = fp.read().strip('\n').split('\n')
+    for line in lines:
+        arr = line.split('\t')
+        key = arr[0]
```

Comment:
You are right. But want to keep it as `OrderedDict` because we can keep the order of subitems (e.g. `frac_reads_in_nfr`, `frac_reads_in_nfr_qc_pass`, `frac_reads_in_nfr_qc_reason` at least, and all of the parser functions `parse_*` in the script return `OrderedDict` instead of `dict`.
```
    for line in lines:
        arr = line.split('\t')
        key = arr[0]
        if key == 'Fraction of reads in NFR':
            result['frac_reads_in_nfr'] = to_float(arr[2])
            result['frac_reads_in_nfr_qc_pass'] = to_bool(arr[1])
            result['frac_reads_in_nfr_qc_reason'] = arr[3]
```